tests/typescript_decorators/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -39,7 +39,10 @@ class AngularComponent {
 }
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 export class TestTextFileService {
-  constructor(@ILifecycleService lifecycleService) {}
+  constructor(
+    @ILifecycleService
+    lifecycleService
```

Comment:
What does this syntax mean?

Reply:
[Here’s what TS has to say](http://www.typescriptlang.org/docs/handbook/decorators.html#parameter-decorators):

> The expression for the parameter decorator will be called as a function at runtime, with the following three arguments:
>
> 1. Either the constructor function of the class for a static member, or the prototype of the class for an instance member.
> 1. The name of the member.
> 1. The ordinal index of the parameter in the function’s parameter list.

Reply:
Interesting. Their example makes it a bit clearer: